static/js/services/enketo.js

```diff
@@ -63,50 +65,55 @@ angular.module('inboxServices').service('Enketo',
     };
 
     var transformXml = function(doc) {
-      return $q
-        .all([
-          XSLT.transform('openrosa2html5form.xsl', doc),
-          XSLT.transform('openrosa2xmlmodel.xsl', doc),
-        ])
-        .then(function(results) {
-          var result = {
-            html: $(results[0]),
-            model: results[1]
-          };
-          return result;
-        });
+      return $q.all([
+        XSLT.transform('openrosa2html5form.xsl', doc),
+        XSLT.transform('openrosa2xmlmodel.xsl', doc),
+      ])
+      .then(function(results) {
+        return {
+          html: $(results[0]),
+          model: results[1]
+        };
+      });
     };
 
-    var xmlCache = {};
+    var translateXml = function(text, title, language) {
+      var xml = $.parseXML(text);
+      var $xml = $(xml);
+      // set the user's language as default so it'll be used for itext translations
+      $xml.find('model itext translation[lang="' + language + '"]').attr('default', '');
+      // manually translate the title as itext doesn't seem to work
+      $xml.find('h\\:title,title').text(TranslateFrom(title));
+      return xml;
+    };
 
-    var withFormByFormInternalId = function(formInternalId) {
+    var getFormXml = function(form, language) {
+      return DB().getAttachment(form.id, FORM_ATTACHMENT_NAME)
+        .then(FileReader)
+        .then(function(text) {
+          return translateXml(text, form.doc.title, language);
+        });
+    };
+
+    var withFormByFormInternalId = function(formInternalId, language) {
       if (!xmlCache[formInternalId]) {
-        xmlCache[formInternalId] = DB()
+        xmlCache[formInternalId] = {};
+      }
+      if (!xmlCache[formInternalId][language]) {
```

Comment:
Not totally convinced we need the `[language]` subsection - doesn't the whole page have to refresh if you change language in the app?